build_locally.sh

```diff
@@ -18,12 +27,22 @@ gem install jekyll
 gem install bundler -v 1.17.3
 bundle install
 
-# Build website using similar script as Travis, but skip htmlproofer because it fails locally:
-bundle exec jekyll build --strict_front_matter
+# Test website using same script as Travis
+if have_travis "$@"; then
```

Comment:
Directy test for existence of .travis.sh script:
```suggestion
if [ -x .travis.sh ] ; then
```

Reply:
Shouldn't there be an `else` branch maintaining previous behaviour?

Reply:
This is not the intended behavior, though. The travis.sh file should always exists. The point of this is that when you are developing locally, you may or may not want to run the full set of tests that travis does. Often I do not. This gives you that option.

Reply:
ive improved the comments to reflect this.

Reply:
Ok, if your objective is to selectively run the travis tests (when the cmdline option `travis` is present), then I suggest something along this line:

```
# Optionally run full travis test (when the cmdline option travis is present)
test [[ " $@ " =~ " travis " ]] && ./.travis.sh
```

Using a function called `have_travis`, suggests that you want to check for the existence of travis, not for the presence of a cmdline argument.

Reply:
using functions like this makes for cleaner code, since a for-loop is necessary once more than one command line argument is needed (future proofing). this ``have_x`` standard i've observed in very mature code bases, and since we don't have a standard i thought it would be good to use as well. how about ``have_x_arg`` then?

note this is just a developer tool i have been improving over the years, and i don't think we need to over dissect it.

Reply:
My one-liner does handle multiple arguments as well: It just looks for the string ` travis ` in the concatenation of _all_ arguments. However, I agree that it is a little more hacky, as I cannot distinguish between `travis` and `"travis is nice"`, i.e. multi-word arguments also containing `travis`.
As you said, this is a developer tool and it should just work. But, we should take care not to confuse the future developer reading this script. And I was confused about the name. If you really insist on your verbose function, please call it `have_travis_arg` as proposed by you.